inc/query/query.php

```diff
@@ -152,4 +153,118 @@ public function handle_query_sibling( &$clauses, $id_column ) {
 
 		$clauses['where'] .= empty( $clauses['where'] ) ? $where : " AND $where";
 	}
+
+	/**
+	 * Modify query join & where statement for multi-relationship.
+	 *
+	 * @param string $clauses   Query clauses.
+	 * @param array  $args   $WP_query args object.
+	 */
+	public function filter_query_clauses( &$clauses, $args ) {
+		global $wpdb;
+		if ( 1 >= count( $args ) ) {
+			return;
+		}
+		$relationships          = $args;
+		$relationships_operator = $clauses['relation'];
+		$join_on_clauses        = array();
+		$in_post_id             = array();
+		foreach ( $relationships as $relationship ) {
+
+			$post_id = $this->get_posts_id( $relationship );
+			if ( empty( $post_id ) || 'OR' === $relationships_operator ) {
+				$in_post_id = array_merge( $in_post_id, $post_id );
+			} else {
+				$in_post_id = array_intersect( $in_post_id, $post_id );
+			}
+			$join_on_clause = $this->get_join_clause( $relationship );
+			if ( null !== $join_on_clause ) {
+				$join_on_clauses[] = $join_on_clause;
+			}
+		}
+		if ( $in_post_id ) {
+			$this_statement_sql = implode( ',', $in_post_id );
+			$this_statement_sql = " AND $wpdb->posts.ID IN(${this_statement_sql})";
+			$clauses['where']  .= $this_statement_sql;
+		}
```

Comment:
Những cái này em viết chỉ cho posts à? Còn terms và users thì thế nào?

Reply:
user và post thì có thể lấy ra post_id ngay và đặt vào WHERE IN(...) còn term thì phải có cái term_id=... AND post IN (...), cái này em nghĩ có thể gom lại và cho term thì thêm term_id=...

Reply:
Có thể gom lại được thành dạng `WHERE $id_column IN (...)` được ko?